tiledb/sm/buffer/buffer.cc

```diff
@@ -154,7 +150,7 @@ bool Buffer::owns_data() const {
   return owns_data_;
 }
 
-Status Buffer::read(void* buffer, uint64_t nbytes) {
+Status Buffer::read(void* buffer, const uint64_t nbytes) {
```

Comment:
I'm all for const-correctness, so this is a welcome change. Let's take it a step further and const the pointer address (NOT the pointer value) with `void *const buffer`, same elsewhere you're making `const` changes.

Reply:
I got the point @joe-maley. but let's make it in another pull request to keep this one simple. What do you think? I will add this TODO in my list and create another one in a few days later.